### PR TITLE
Add types field to package.json

### PR DESCRIPTION
**Pull request by G-Rath**
This is the typescript equivalent of `main`.

Without it, you'll generally get IDEs defaulting to import from `mock-stdin/lib` rather than just `mock-stdin`.

Alternatively, you can have the `index.d.ts` at the root of the project.

### CODE REVIEW

**Comment by caitp** [comment]
If you wouldn’t mind terribly, it might be good to move the type definitions into a separate directory. That will solve some other problems in the CI script. Otherwise, lgtm

**Comment by G-Rath** [comment]
No problem, any particular separate directory?

However, I wouldn't expect the types to be causing CI troubles - Feel free to chuck a link or dump of any specific troubles you've having if you'd like me to have a look at fixing them :)

**Comment by caitp** [comment]
The issue is the type definition file breaks jscoverage, and the exclude pattern option in that package is broken, so instrumenting the library for coverage is broken. It would be fixed by moving the type definitions to a separate directory (like root/types instead of root/lib), or just in the root of the repo itself.

File: package.json [hunk 1]
```diff
@@ -3,6 +3,7 @@
   "version": "1.0.0",
   "description": "Mock STDIN file descriptor in Node.js",
   "main": "./lib/index.js",
+  "types": "./lib/index.d.ts",
```
Comment (caitp) [inline package.json]: I think we still need to update this, right?

Comment (G-Rath) [reply package.json]: pffft yeah silly me!

Comment (caitp) [reply package.json]: I meant, to point to the file in the root directory. I could be wrong, not an expert in the typescript stuff, so if it’s fine lgtm

Comment (G-Rath) [reply package.json]: > I could be wrong

You're not - that was my literal reaction when I saw your comment & the line, b/c that was the point of the PR in the first place and so couldn't believe I'd forgotten to update it 😹 

I've amended my commit to change that, so it's fixed now; good catch 😄

**Comment by caitp** [comment]
Thanks so much

**Pull request merged**